qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -951,33 +956,58 @@ def _update_individual_axes_ramp_rates(
+        with ExitStack() as restore_ramp_rates_stack:
+            for instrument in (
+                self._instrument_x,
+                self._instrument_y,
+                self._instrument_z,
+            ):
+                restore_ramp_rates_stack.enter_context(
+                    instrument.ramp_rate.restore_at_exit()
+                )
+            restore_ramp_rates_stack.callback(
+                self.log.debug,
+                "Simultaneous ramp: restoring individual axes ramp rates",
+            )
+
+            self._update_individual_axes_ramp_rates(values)
 
-        axes = (self._instrument_x, self._instrument_y, self._instrument_z)
+            axes = (self._instrument_x, self._instrument_y, self._instrument_z)
 
-        for axis_instrument, value in zip(axes, values):
-            current_actual = axis_instrument.field()
+            for axis_instrument, value in zip(axes, values):
+                current_actual = axis_instrument.field()
+
+                # If the new set point is practically equal to the
+                # current one then do nothing
+                if np.isclose(value, current_actual, rtol=0, atol=1e-8):
+                    self.log.debug(
+                        f"Simultaneous ramp: {axis_instrument.short_name} is "
+                        f"already at target field {value} "
+                        f"{axis_instrument.field.unit} "
+                        f"({current_actual} exactly)"
+                    )
+                    continue
 
-            # If the new set point is practically equal to the
-            # current one then do nothing
-            if np.isclose(value, current_actual, rtol=0, atol=1e-8):
                 self.log.debug(
-                    f"Simultaneous ramp: {axis_instrument.short_name} is "
-                    f"already at target field {value} "
-                    f"{axis_instrument.field.unit} "
-                    f"({current_actual} exactly)"
+                    f"Simultaneous ramp: setting {axis_instrument.short_name} "
+                    f"target field to {value} {axis_instrument.field.unit}"
+                )
+                axis_instrument.set_field(
+                    value, perform_safety_check=False, block=False
                 )
-                continue
-
-            self.log.debug(
-                f"Simultaneous ramp: setting {axis_instrument.short_name} "
-                f"target field to {value} {axis_instrument.field.unit}"
-            )
-            axis_instrument.set_field(value, perform_safety_check=False, block=False)
 
-        if self.block_during_ramp() is True:
-            self.log.debug(f"Simultaneous ramp: blocking until ramp is finished")
-            self.wait_while_all_axes_ramping()
+            if self.block_during_ramp() is True:
+                self.log.debug(f"Simultaneous ramp: blocking until ramp is finished")
+                self.wait_while_all_axes_ramping()
```

Comment:
I know this code is just moved but I am surprised to see this log before waiting

Reply:
hmm, that log message was there before, the "blocking until ramp is finished" one. so what is surprising and why?

Reply:
nevermind misread